### ISSUE TITLE
Ambiguous method error running FlowDroid

### ISSUE THREAD

**Issue by zhanghan177** [secure-software-engineering/FlowDroid#82]
Hi,

I encountered this error when running FlowDroid with the following APK: https://github.com/zhanghan177/file_share/blob/master/com.mobincube.android.sc_337MSK.apk

FlowDroid failed to calculate callback methods, as is shown in the log
```
[main] INFO soot.jimple.infoflow.android.SetupApplication - Constructing the callgraph...
[main] INFO soot.jimple.infoflow.android.callbacks.DefaultCallbackAnalyzer - Collecting callbacks in DEFAULT mode...
[main] INFO soot.jimple.infoflow.android.callbacks.DefaultCallbackAnalyzer - Callback analysis done.
...
[main] ERROR soot.jimple.infoflow.android.SetupApplication - Could not calculate callback methods
java.lang.RuntimeException: ambiguous method: dummyMainMethod_AdActivity in class dummyMainClass
	at soot.SootClass.getMethodByNameUnsafe(SootClass.java:606)
	at soot.jimple.infoflow.android.entryPointCreators.components.AbstractComponentEntryPointCreator.createEmptyMainMethod(AbstractComponentEntryPointCreator.java:88)
	at soot.jimple.infoflow.entryPointCreators.BaseEntryPointCreator.createDummyMain(BaseEntryPointCreator.java:142)
	at soot.jimple.infoflow.android.entryPointCreators.AbstractAndroidEntryPointCreator.createDummyMain(AbstractAndroidEntryPointCreator.java:25)
	at soot.jimple.infoflow.android.entryPointCreators.AndroidEntryPointCreator.createDummyMainInternal(AndroidEntryPointCreator.java:254)
	at soot.jimple.infoflow.entryPointCreators.BaseEntryPointCreator.createDummyMain(BaseEntryPointCreator.java:154)
	at soot.jimple.infoflow.android.entryPointCreators.AbstractAndroidEntryPointCreator.createDummyMain(AbstractAndroidEntryPointCreator.java:25)
	at soot.jimple.infoflow.android.SetupApplication.createMainMethod(SetupApplication.java:1008)
	at soot.jimple.infoflow.android.SetupApplication.calculateCallbackMethods(SetupApplication.java:664)
	at soot.jimple.infoflow.android.SetupApplication.calculateCallbacks(SetupApplication.java:477)
	at soot.jimple.infoflow.android.SetupApplication.calculateCallbacks(SetupApplication.java:447)
	at soot.jimple.infoflow.android.SetupApplication.processEntryPoint(SetupApplication.java:1392)
	at soot.jimple.infoflow.android.SetupApplication.runInfoflow(SetupApplication.java:1361)
	at soot.jimple.infoflow.android.SetupApplication.runInfoflow(SetupApplication.java:1307)
	at soot.jimple.infoflow.cmd.MainClass.run(MainClass.java:237)
	at soot.jimple.infoflow.cmd.MainClass.main(MainClass.java:196)
```

This analysis quit quite fast at an early stage. Any idea what might be causing this error? 

BTW, VirusTotal reports this APK as an Adware. Please be aware. (https://www.virustotal.com/#/file/b337bf1988e2201ba76b431761b417b845d6310869f05bab31db76b36d908f49/detection)

**Comment by nanjingdaqi** [comment]
The issue should has been fixed by commit https://github.com/secure-software-engineering/FlowDroid/commit/72734bd629dfae2aacaf6e6973abfe73d035c979.

The reason is component's dummy_entry_method's name is constructed only by component's simple class name. And there are more than one AdActivity with different package name.

**Comment by StevenArzt** [comment]
I'll close this issue, because it should be fixed in version 2.7. Please report if the problem still persists.
